src/screens/CreatePost/index.js

```diff
@@ -506,14 +509,23 @@ const CreatePost = () => {
         });
       }
 
-      await createPost(data);
+      const post = await createPost(data);
       if (params.onRefresh && typeof params.onRefresh === 'function') {
         params.onRefresh();
       }
-      showMessage({
-        message: StringConstant.createPostDone,
-        type: 'success'
-      });
+      if (post.code === 200) {
+        showMessage({
+          message: StringConstant.createPostDone,
+          type: 'success'
+        });
+        setLoadingPost(false);
+      } else {
+        showMessage({
+          message: StringConstant.createPostFailedGeneralError,
+          type: 'danger'
+        });
+        setLoadingPost(false);
+      }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The changes made here are significant for better error handling and user experience. The `createPost` function is now awaited and its response is checked. If the post creation is successful, a success message is shown and `loadingPost` is set to false. If it fails, an error message is shown and `loadingPost` is also set to false. This ensures that the loading state is properly managed regardless of the outcome of the post creation.

```javascript
const post = await createPost(data);
if (params.onRefresh && typeof params.onRefresh === 'function') {
  params.onRefresh();
}
if (post.code === 200) {
  showMessage({
    message: StringConstant.createPostDone,
    type: 'success'
  });
  setLoadingPost(false);
} else {
  showMessage({
    message: StringConstant.createPostFailedGeneralError,
    type: 'danger'
  });
  setLoadingPost(false);
}
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->